news/409.bugfix

```diff
@@ -0,0 +1 @@
+Assigning a primitive type to a Subscripted Dict now raises a descriptive message.
```

Comment:
This is two different bugs.
1. Bad error message
2. Mutating the container on exception

I would say that the second one is actually much worse, but you seems to ignore it here.

Reply:
Do you want another diff for this ? Or do I change this news fragment?

Reply:
No need for another diff, but create two issues and have two news fragments in your diff.

Reply:
Can you add a second news fragment?
If you want to use the same issue you can name them 409.bugfix.1 and 409.bugfix.2.